docs/src/reference/architecture.md

```diff
@@ -0,0 +1,72 @@
+# Architecture
+
+A system architecture document is the starting point for many interested participants in a project, whether you intend contirbuting or simply want to understand how the software is structured. This documentation lays out the current design of Canonical Kubernetes, following the [C4 model]. 
+
+##  System context 
+
+This overview of Canononical Kubernetes demonstrates the interactions of Kubernetes with users and with other systems.
+
+```{kroki} ../assets/overview.puml
+```
+
+Two actors interact with the kubernetes snap:
+
+- **K8s admin**: The administrator of the cluster interacts directly with the Kubernetes API server. Out of the box our K8s distribution offers admin access to the cluster. That initial user is able to configure the cluster to match their needs and of course create other users that may or may not have the same privileges. The K8s admin is also able to deploy workloads running in the cluster.
```

Comment:
maybe `maintain` instead of `deploy` workloads? An admin can e.g. also patch and remove workloads